testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MainActivity.java

```diff
@@ -233,26 +233,47 @@ public void onAcquireTokenClicked(final AcquireTokenFragment.RequestOptions requ
         callAcquireToken(mScopes, mUiBehavior, mLoginHint, mExtraQp, mExtraScopesToConsent);
     }
 
-    @Override
     public void onRemoveUserClicked() {
-        if (mUser == null) {
-            showMessage("Please select a user");
-            return;
+        try {
+            final List<User> users = mApplication.getUsers();
+            for (final User user : users) {
+                mApplication.remove(user);
+            }
+        } catch (final MsalClientException e) {
+            Log.e(TAG, "Fail to retrieve users: " + e.getMessage(), e);
         }
 
-        mApplication.remove(mUser);
         mUser = null;
     }
 
+    User getUser(String loginHint) {
+        try {
+            final List<User> users = mApplication.getUsers();
+            for (final User user : users) {
+                if (user.getDisplayableId().equals(loginHint.trim().toLowerCase())) {
+                    return user;
+                }
+            }
+        } catch (final MsalClientException e) {
+            Log.e(TAG, "Fail to retrieve users: " + e.getMessage(), e);
+        }
+
+        showMessage("Users are removed.");
```

Comment:
Can we call `showMessage()` conditionally only if the `User` was _actually_ removed? Maybe consider showing an error message if the Exception is thrown or if the `login_hint` can't be found?